File: Binary_search/Low_bound.py
def low_bound(arr,target,start,end,cur=0):
  #target과 같거나 큰 idx가 처음으로 나오는 위치를 return  
  if start>=end:
    return cur
  else:
    mid=int((start+end)/2)
    if arr[mid]<target:
      return low_bound(arr,target,mid+1,end,cur)
    elif arr[mid]>=target:
      cur=mid
      return low_bound(arr,target,start,mid,cur)

if __name__=='__main__':
  sorted_array=[10, 10, 10, 13, 20, 20 ,20]
  print(low_bound(sorted_array,20,0,len(sorted_array)))

File: Binary_search/Low_bound.cpp
#include<bits/stdc++.h>
using namespace std;

int low_bound_binary_search(int *arr, int target, int start ,int end){
  while(start<end)
  { 
    int mid=(start+end)/2; 
    if(arr[mid]<target){
      start=mid+1;
    }
    else if(arr[mid]>=target){
      end=mid;
    }
  }
  return end;
}

int main(){
  int array[11]={1,1,1,2,2,2,3,3,3,4,6};
  int target;
  cin>>target;
  cout<<low_bound_binary_search(array,target,0,sizeof(array)/sizeof(int)-1);
  system("pause");
}
